lab_5_scraper/scraper.py

```diff
@@ -206,12 +457,45 @@ def prepare_environment(base_path: Union[pathlib.Path, str]) -> None:
     Args:
         base_path (Union[pathlib.Path, str]): Path where articles stores
     """
+    path = pathlib.Path(base_path)
+    if path.is_dir():
+        shutil.rmtree(base_path)
+    path.mkdir(parents=True)
 
 
 def main() -> None:
     """
     Entrypoint for scrapper module.
     """
+    prepare_environment(ASSETS_PATH)
+    config = Config(CRAWLER_CONFIG_PATH)
+    crawler = Crawler(config)
+    crawler.find_articles()
+    for idx, url in enumerate(crawler.urls, 1):
+        parser = HTMLParser(url, idx, config)
+        article = parser.parse()
+        if isinstance(article, Article):
+            to_raw(article)
+            to_meta(article)
+
+
+def main_recursive_crawler() -> None:
+    """
+    Recursive crawler showcase.
+    """
+    prepare_environment(ASSETS_PATH)
```

Comment:
I tried your parser for the third time (final attempt) with removed condition to stop if number of urls exceeds the value specified in config, and it starts obtaining duplicate values, see attached.

[recursive_crawler_cache.json](https://github.com/user-attachments/files/20015726/recursive_crawler_cache.json)